### ISSUE TITLE
Issue with AWSMobileClient.getTokens with custom endpoint

### ISSUE THREAD

**Issue by Rusik** [aws-amplify/aws-sdk-ios#4153]
Hey guys

Recently I added support of custom endpoint to Amplify iOS: https://github.com/aws-amplify/amplify-ios/pull/1715
But when I tried to use it I found an issue and seems it is on aws sdk side.

The issue is when I call `Amplify.Auth.fetchAuthSession()` which calls [AWSMobileClient.getTokens()](https://github.com/aws-amplify/aws-sdk-ios/blob/0b1bf2515c42c73777f941066f8526a4bac92fc7/AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClientOperations/AWSMobileClient%2BTokens.swift#L29) under the hood, I get an error like
```
AWSAuthCognitoSession(isSignedIn: true, userSubResult: Swift.Result<Swift.String, Amplify.AuthError>.success("7a937fde-d2d0-4e49-85e6-335984bb3870"), identityIdResult: Swift.Result<Swift.String, Amplify.AuthError>.failure(AuthError: Could not fetch identity Id, AWS Cognito Identity Pool is not configured
Recovery suggestion: Follow the steps to configure AWS Cognito Identity Pool and try again
Caused by:
invalidAccountTypeException), awsCredentialsResult: Swift.Result<AWSPluginsCore.AuthAWSCredentials, Amplify.AuthError>.failure(AuthError: Could not fetch AWS Credentials, AWS Cognito Identity Pool is not configured
Recovery suggestion: Follow the steps to configure AWS Cognito Identity Pool and try again
Caused by:
invalidAccountTypeException), cognitoTokensResult: Swift.Result<AWSPluginsCore.AuthCognitoTokens, Amplify.AuthError>.success(AmplifyPlugins.AWSCognitoUserPoolTokens(idToken: "<hidden>", accessToken: "<hidden>", refreshToken: "<hidden>")))
```

On android everything worked fine so I tried to look at the network traffic and found a difference

When android asks for `X-Amz-Target: AWSCognitoIdentityService.GetId` it uses `cognito-idp.eu-west-2.amazonaws.com/eu-west-2_W5sRvOSkW` for key inside Body
<img width="1552" alt="Screenshot 2022-05-24 at 18 34 02" src="https://user-images.githubusercontent.com/800482/170012511-67be78aa-d47b-4c14-8c63-56ce1049d5fb.png">

But when iOS attempts to request the same method, it uses key with custom endpoint prefix `auth.agoraworld.dev/eu-west-2_W5sRvOSkW` and it fails

<img width="1552" alt="Screenshot 2022-05-24 at 18 32 57" src="https://user-images.githubusercontent.com/800482/170012776-3b345e70-afe8-4831-ab4d-76a058420551.png">

If I retry ios request with the key from android request everything works fine.

I found the code where this key is creating and seems like it should be implemented the way it work on android

On android it looks like this

[link to the code](https://github.com/aws-amplify/aws-sdk-android/blob/8de48797ae2e5c157169a94746c768da7de61056/aws-android-sdk-auth-userpools/src/main/java/com/amazonaws/mobile/auth/userpools/CognitoUserPoolsSignInProvider.java#L380)
```java
cognitoLoginKey = "cognito-idp." + getValueFromConfig("Region") + ".amazonaws.com/" + cognitoUserPool.getUserPoolId();
```

[link to the code](https://github.com/aws-amplify/aws-sdk-android/blob/7274541eae44f37b7ba0d8e16b8e886fee922c38/aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java#L603)
```java
userpoolsLoginKey = String.format("cognito-idp.%s.amazonaws.com/%s", userPoolJSON.getString("Region"), userPoolJSON.getString("PoolId"));
```

While on iOS uses custom endpoint instead of `cognito-idp.%s.amazonaws.com` 
[link to the code](https://github.com/aws-amplify/aws-sdk-ios/blob/d0ad8ac3516722a7271d8a2ae42da68e2de29478/AWSCognitoIdentityProvider/AWSCognitoIdentityUserPool.m#L297)
```obj-c
[NSString stringWithFormat:@"%@/%@", self.client.configuration.endpoint.hostName, self.userPoolConfiguration.poolId];
```

So if I am right, it should be something like on iOS code
```objc
[NSString stringWithFormat:@"cognito-idp.%@.amazonaws.com/%@", self.configuration.endpoint.regionName, self.userPoolConfiguration.poolId];
```

**Comment by royjit** [comment]
Thank you for opening the issue and the detailed analysis of the issue. I have created this PR to fix it: https://github.com/aws-amplify/aws-sdk-ios/pull/4162

**Comment by Rusik** [comment]
@royjit thank you very much! 
Do you know how soon release with this PR will be available?

**Comment by royjit** [comment]
This is released today  as version `2.27.10` - https://github.com/aws-amplify/aws-sdk-ios/releases/tag/2.27.10
